hathor/verification/merge_mined_block_verifier.py

```diff
@@ -12,14 +12,29 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+from hathor.conf.settings import HathorSettings
+from hathor.feature_activation.feature import Feature
+from hathor.feature_activation.feature_service import FeatureService
 from hathor.transaction import MergeMinedBlock
+from hathor.transaction.aux_pow import NEW_MAX_MERKLE_PATH_LENGTH, OLD_MAX_MERKLE_PATH_LENGTH
 
 
 class MergeMinedBlockVerifier:
-    __slots__ = ()
+    __slots__ = ('_settings', '_feature_service',)
+
+    def __init__(self, *, settings: HathorSettings, feature_service: FeatureService):
+        self._settings = settings
+        self._feature_service = feature_service
 
     def verify_aux_pow(self, block: MergeMinedBlock) -> None:
         """ Verify auxiliary proof-of-work (for merged mining).
         """
         assert block.aux_pow is not None
-        block.aux_pow.verify(block.get_base_hash())
+
+        is_feature_active = self._feature_service.is_feature_active(
+            block=block,
+            feature=Feature.INCREASE_MAX_MERKLE_PATH_LENGTH
+        )
+        max_merkle_path_length = NEW_MAX_MERKLE_PATH_LENGTH if is_feature_active else OLD_MAX_MERKLE_PATH_LENGTH
```

Comment:
Already mentioned previously, but I'd prefer this value to be in settings, we then could have different values for the testnet/mainnet/privatenets, not that it is useful in practice, but it will be useful for starting the experiment on the testnet with a non-optimal value, and choosing an optimal value for the mainnet without interfering with the testnet.

Reply:
Makes sense, changed in https://github.com/HathorNetwork/hathor-core/pull/910/commits/f4b6182ad58640dc0a0f51cdeb967e178c298873